packages/circuit-ui/components/ModalContext/ModalContext.tsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright 2019, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useCallback,
+  useDebugValue,
+  ReactNode,
+  MouseEvent,
+  KeyboardEvent,
+} from 'react';
+import ReactModal, { Props as ReactModalProps } from 'react-modal';
+import { Global, css } from '@emotion/core';
+import { Dispatch as TrackingProps } from '@sumup/collector';
+
+import { useStack, StackItem, StackDispatch } from '../../hooks/useStack';
+import { uniqueId } from '../../util/id';
+
+// It is important for users of screenreaders that other page content be hidden
+// (via the `aria-hidden` attribute) while the modal is open.
+// To allow react-modal to do this, Circuit UI calls `Modal.setAppElement`
+// with a query selector identifying the root of the app.
+// http://reactcommunity.org/react-modal/accessibility/#app-element
+if (typeof window !== 'undefined') {
+  // These are the default app elements in Next.js and CRA.
+  const appElement =
+    document.getElementById('__next') || document.getElementById('root');
+
+  if (appElement) {
+    ReactModal.setAppElement(appElement);
+  } else if (
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test'
+  ) {
+    console.error(
+      [
+        '[ModalProvider] Could not find the app root element to hide it',
+        'when a modal is open. Add an element with the id `#root`',
+        'at the root of your application to remove this error.',
```

Comment:
Should we make this configurable so that any id could be passed? Something like `rootId`?

Reply:
This is outside the component (has to be), so, unfortunately, making it configurable is not possible.